### PR TITLE
Initial working CCM prototype

### PR DESCRIPTION
**Pull request by frapposelli**
This is an initial stab at code refactoring, the provider is registering nodes successfully but code requires a lot of cleanup. It also imports code from in-tree as #7 requires.

### CODE REVIEW

**Comment by frapposelli** [comment]
tested positively on kubernetes 1.11

**Comment by dvonthenen** [comment]
@frapposelli yep, I have been providing feedback to you via slack. I will take another pass at this with your latest changes and comment here.

**Comment by dvonthenen** [comment]
@frapposelli Other than the filename change below, everything else looks fine. We have issues reflecting the work left to do as well as comments in the code with TODO as placeholders. This is a pretty good start we can begin iterating on. Awesome!

Let's change the following file names from:
manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml
manifests/controller-manager/openstack-cloud-controller-manager-pod.yaml

To:
vsphere-cloud-controller-manager-ds.yaml
vsphere-cloud-controller-manager-pod.yaml

**Comment by dvonthenen** [comment]
Also, we might want to delete the file pkg/vclib/vsphere_metrics.go. Those metrics are specific to the CSI driver and probably no longer need to be among shared code.

File: Gopkg.toml [hunk 1]
```diff
@@ -38,23 +46,23 @@ ignored = ["k8s.io/cloud-provider-vsphere"]
 
 [[override]]
   name = "k8s.io/api"
-  version = "kubernetes-1.10.0"
+  version = "kubernetes-1.11.0"
```
[the remark this answers is not in the document]
Comment (tpepper) [reply Gopkg.toml]: Definitely!

**Pull request merged**